Lib/traceback.py

```diff
@@ -532,6 +503,47 @@ def __init__(self, exc_type, exc_value, exc_traceback, *, limit=None,
             self.msg = exc_value.msg
         if lookup_lines:
             self._load_lines()
+        self.__suppress_context__ = \
+            exc_value.__suppress_context__ if exc_value else False
+
+        # Convert __cause__ and __context__ to `TracebackExceptions`s, use a
+        # queue to avoid recursion
```

Comment:
We could use an additional comment reminding the reader that the recursion here is only one level deep because the `_seen` we pass is never None, so the recursive invocations below always set `is_chained` which stops the recursion. (It took me way too long to figure this out without help. :-)